test/e2e_node/eviction_test.go

```diff
@@ -0,0 +1,601 @@
+							Name:  podName,
+						},
+					},
+				},
+			})
+		})
+
+		AfterEach(func() {
+			By("deleting pods")
+			for _, spec := range testSpecs {
+				By(fmt.Sprintf("deleting pod: %s", spec.pod.Name))
+				f.PodClient().DeleteSync(spec.pod.Name, &metav1.DeleteOptions{}, longPodDeletionTimeout)
+			}
+
+			if CurrentGinkgoTestDescription().Failed {
+				if framework.TestContext.DumpLogsOnFailure {
+					logPodEvents(f)
+					logNodeEvents(f)
+				}
+				By("sleeping to allow for cleanup of test")
+				time.Sleep(postTestConditionMonitoringPeriod)
+			}
+		})
+	})
+}
+
+// verifyEvictionOrdering returns an error if all non-zero priority pods have not been evicted, nil otherwise
+// This function panics (via Expect) if eviction ordering is violated, or if a priority-zero pod fails.
+func verifyEvictionOrdering(f *framework.Framework, testSpecs []podEvictSpec) error {
+	// Gather current information
+	updatedPodList, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	updatedPods := updatedPodList.Items
+	for _, p := range updatedPods {
+		framework.Logf("fetching pod %s; phase= %v", p.Name, p.Status.Phase)
+	}
+
+	By("checking eviction ordering and ensuring important pods dont fail")
+	done := true
+	for _, priorityPodSpec := range testSpecs {
+		var priorityPod v1.Pod
+		for _, p := range updatedPods {
+			if p.Name == priorityPodSpec.pod.Name {
+				priorityPod = p
+			}
+		}
+		Expect(priorityPod).NotTo(BeNil())
+
+		// Check eviction ordering.
+		// Note: it is alright for a priority 1 and priority 2 pod (for example) to fail in the same round
+		for _, lowPriorityPodSpec := range testSpecs {
+			var lowPriorityPod v1.Pod
+			for _, p := range updatedPods {
+				if p.Name == lowPriorityPodSpec.pod.Name {
+					lowPriorityPod = p
+				}
+			}
+			Expect(lowPriorityPod).NotTo(BeNil())
+			if priorityPodSpec.evictionPriority < lowPriorityPodSpec.evictionPriority && lowPriorityPod.Status.Phase == v1.PodRunning {
+				Expect(priorityPod.Status.Phase).NotTo(Equal(v1.PodFailed),
+					fmt.Sprintf("priority %d pod: %s failed before priority %d pod: %s",
+						priorityPodSpec.evictionPriority, priorityPodSpec.pod.Name, lowPriorityPodSpec.evictionPriority, lowPriorityPodSpec.pod.Name))
+			}
+		}
+
+		// EvictionPriority 0 pods should not fail
+		if priorityPodSpec.evictionPriority == 0 {
+			Expect(priorityPod.Status.Phase).NotTo(Equal(v1.PodFailed),
+				fmt.Sprintf("priority 0 pod: %s failed", priorityPod.Name))
+		}
+
+		// If a pod that is not evictionPriority 0 has not been evicted, we are not done
+		if priorityPodSpec.evictionPriority != 0 && priorityPod.Status.Phase != v1.PodFailed {
+			done = false
+		}
+	}
+	if done {
+		return nil
+	}
+	return fmt.Errorf("pods that should be evicted are still running")
+}
+
+// Returns TRUE if the node has the node condition, FALSE otherwise
+func hasNodeCondition(f *framework.Framework, expectedNodeCondition v1.NodeConditionType) bool {
```

Comment:
Still prefer splitting the logging and predicate into separate functions, something like:
```
func hasNodeCondition(condition v1.NodeConditionType) bool {...}
func logNodeCondition(f *framework.Framework, condition v1.NodeConditionType, expected bool) {...}
```

Reply:
done